pkg/provisioner/ironic/testserver/redfish.go

```diff
@@ -0,0 +1,703 @@
+package testserver
+
+import (
+	"net/http"
+	"testing"
+)
+
+// RedfishMock is a test server that implements Redfish Virtual Media semantics
+type RedfishMock struct {
+	*MockServer
+}
+
+// NewRedfish builds a redfish virtual media mock server
+func NewRedfish(t *testing.T) *RedfishMock {
+	return &RedfishMock{
+		MockServer: NewTLS(t, "redfish"),
+	}
+}
+
+// Endpoint returns the URL to the server
+func (m *RedfishMock) Endpoint() string {
+	return m.MockServer.RawEndpoint()
+}
+
+// RootService configures the server with a valid response for /redfish/v1/
+func (m *RedfishMock) rootService() *RedfishMock {
+	m.ResponseWithCode("/redfish/v1/", `
+	{
+		"@odata.type": "#ServiceRoot.v1_9_0.ServiceRoot",
+		"Id": "RootService",
+		"Name": "Root Service",
+		"RedfishVersion": "1.6.0",
+		"UUID": "92384634-2938-2342-8820-489239905423",
+		"Systems": {
+			"@odata.id": "/redfish/v1/Systems"
+		},
+		"Chassis": {
+			"@odata.id": "/redfish/v1/Chassis"
+		},
+		"Managers": {
+			"@odata.id": "/redfish/v1/Managers"
+		},
+		"Tasks": {
+			"@odata.id": "/redfish/v1/TaskService"
+		},
+		"SessionService": {
+			"@odata.id": "/redfish/v1/SessionService"
+		},
+		"AccountService": {
+			"@odata.id": "/redfish/v1/AccountService"
+		},
+		"EventService": {
+			"@odata.id": "/redfish/v1/EventService"
+		},
+		"Registries": {
+			"@odata.id": "/redfish/v1/Registries"
+		},
+		"UpdateService": {
+			"@odata.id": "/redfish/v1/UpdateService"
+		},
+		"CertificateService": {
+			"@odata.id": "/redfish/v1/CertificateService"
+		},
+		"Links": {
+			"Sessions": {
+				"@odata.id": "/redfish/v1/SessionService/Sessions"
+			}
+		},
+		"Oem": { },
+		"@odata.id": "redfish/v1/"
+	}`, http.StatusOK)
+	return m
+}
+
+// Unauthorized configures the server with a http 401 Unauthorized response for /redfish/v1/
+func (m *RedfishMock) Unauthorized() *RedfishMock {
+	m.ErrorResponse("/redfish/v1/", http.StatusUnauthorized)
+	return m
+}
+
+// NoSystem configures the server with a valid response for /redfish/v1
+func (m *RedfishMock) NoSystem() *RedfishMock {
+	m.rootService()
+	m.ResponseWithCode("/redfish/v1/Systems", `
+	{
+		"@odata.context": "/redfish/v1/$metadata#ComputerSystemCollection.ComputerSystemCollection",
```

Comment:
I guess that also in this case the method could be a lot simpler if using a JSON object like https://github.com/stmcginnis/gofish/blob/1d2c11e698c966d28c6ecf1116d3f2131ab70804/redfish/computersystem.go#L374 (note: also the Ironic mock adopted this pattern, by reusing the types offered by goperhcloud, see [here](https://github.com/metal3-io/baremetal-operator/blob/b46fb6e6edd41f49fa5a96128398cc5a16adff35/pkg/provisioner/ironic/testserver/ironic.go#L107])

Reply:
I see, I will update to receive the ComputerSystem struct.

Reply:
we can't use directly the ComputerSystem struct or Manager struct, the gofish struct when turns into json is not the same json redfish would return, and for Manager I can't set `virtualMedia` information because  it's private in the struct.